x/sysproxy/sysproxy_darwin.go

```diff
@@ -0,0 +1,295 @@
+//	recvpipe  sendpipe  ssthresh  rtt,msec    rttvar  hopcount      mtu     expire
+//	0         0         0         0         0         0      1500         0
+func getDefaultRouteInterface() (string, error) {
+	// Execute a command to get the default route
+	out, err := exec.Command("route", "get", "default").Output()
+	if err != nil {
+		return "", err
+	}
+
+	// Extract the interface name from the command output
+	output := string(out)
+	lines := strings.Split(output, "\n")
+	var defaultIface string
+	for _, line := range lines {
+		if strings.HasPrefix(strings.TrimSpace(line), "interface:") {
+			fields := strings.Fields(line)
+			if len(fields) >= 2 {
+				defaultIface = fields[1]
+				return defaultIface, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("failed to get default route interface")
+}
+
+// getNetworkServiceName parses the output of networksetup -listnetworkserviceorder to find
+// the network service name for a given hardware port (e.g. Wi-Fi for en0)
+func getNetworkServiceName(output, hardwarePort string) (string, error) {
+	const pattern = `Hardware Port: ([^,]+),`
+	// example line: (Hardware Port: Wi-Fi, Device: en0)
+	lines := strings.Split(output, "\n")
+	for _, line := range lines {
+		if strings.Contains(line, hardwarePort) {
+			re := regexp.MustCompile(pattern)
+			matches := re.FindStringSubmatch(line)
+			if len(matches) >= 2 {
+				return matches[1], nil
+			}
+		}
+	}
+	return "", fmt.Errorf("failed to find network service name for hardware port %s", hardwarePort)
+}
+
+// setProxySettings sets the specified type of proxy on the given network interface.
+// https://keith.github.io/xcode-man-pages/networksetup.8.html#getsecurewebproxy
+func setProxySettings(p ProxyType, interfaceName string, host string, port string) error {
+	switch p {
+	case proxyTypeHTTP:
+		return exec.Command("networksetup", "-setwebproxy", interfaceName, host, port).Run()
+	case proxyTypeHTTPS:
+		return exec.Command("networksetup", "-setsecurewebproxy", interfaceName, host, port).Run()
+	case proxyTypeSOCKS:
+		return exec.Command("networksetup", "-setsocksfirewallproxy", interfaceName, host, port).Run()
+	default:
+		return fmt.Errorf("unsupported proxy type: %s", p)
+	}
+}
+
+// disableProxy turns off the specified type of proxy on the given network interface.
+// https://keith.github.io/xcode-man-pages/networksetup.8.html#setwebproxystate
+func disableProxy(p ProxyType, interfaceName string) error {
+	switch p {
+	case proxyTypeHTTP:
+		return exec.Command("networksetup", "-setwebproxystate", interfaceName, "off").Run()
+	case proxyTypeHTTPS:
+		return exec.Command("networksetup", "-setsecurewebproxystate", interfaceName, "off").Run()
+	case proxyTypeSOCKS:
+		return exec.Command("networksetup", "-setsocksfirewallproxystate", interfaceName, "off").Run()
+	default:
+		return fmt.Errorf("unsupported proxy type: %s", p)
+	}
+}
+
+func getProxySettings(p ProxyType, interfaceName string) (*proxySettings, error) {
```

Comment:
I think this is dead code. We can remove the get stuff.

Reply:
@fortuna I am using `getProxySettings` in tests to ensure parameters are properly set by the setProxy action.